src/constants/api.ts

```diff
@@ -1,154 +1,246 @@
+        throw error;
     }
 }
 
+// Absences and presences sorted by Squad
+
 export const getLeavesBySquad = async (squad: Squad) => {
 
     try {
-        console.log(`Bienvenue dans ce nouveau sprint! \nVoici les présences et absences prévues du -- au -- \n`)
-        console.log(`${squad.name}:`);
+        const totalDevelopers = squad.userIds.length
+        let totalPresenceDays = 0
+        let totalAbsenceDays = 0
+        const fullyPresentUsers: string[] = []
+        const absentUsers: AbsentUsers[] = []
+
+        if(squad.userIds.length > 0){
 
         for (const userId of squad.userIds) {
-            const leavesData = await getLeavesByUserId(userId);
-            const absenceDays = leavesData.data.items.length / 2;
-            const presenceDays = totalDays - absenceDays;
+            const leavesData = await getLeavesByUserId(userId)
+            const absenceDays = leavesData?.data?.items?.length / 2
+            const presenceDays = totalDays - absenceDays
 
-            const user = (await allTechUsers()).items.find(user => user.id === userId);
-            if (user) {
-                console.log(`${user.firstName} ${user.lastName} sera présent(e) ${presenceDays} jours sur 10`);
-            }
-            if (user && leavesData.data.items.length > 0) {
-                console.log(`Jours d'absence pour ${user.firstName} ${user.lastName}:`);
-                const allDay: string[] = [];
-                leavesData.data.items.forEach(item => {
-                    allDay.push(item.date);
-                    
-                    const am = item.isAM;
-                });
-                    const uniqueDay = allDay.filter((obj, index, self) => self.findIndex(o => JSON.stringify(o) === JSON.stringify(obj)) === index);
-                    uniqueDay.forEach(day =>{
-                        console.log((dayjs(day).format(`DD/MM/YYYY`)));
+            totalPresenceDays += presenceDays
+            totalAbsenceDays += absenceDays
+
+            const groupedByDate: GroupedByDate = _.groupBy(leavesData.data.items, 'date')
 
+            const user = (await allTechUsers()).items.find(user => user.id === userId)
+            if (user) {
+                if (presenceDays === totalDays) {
+                    fullyPresentUsers.push(`${user.firstName} ${user.lastName}`);
+                } else if (presenceDays < totalDays) {
+                    const absences: string[] = []
+                    for (const date in groupedByDate) {
+                        const items = groupedByDate[date]
+
+                        const amLeave = items.find(item => item.isAM === true)!                       
+                        const pmLeave = items.find(item => item.isAM === false)!
+
+                        const formatDate = dayjs(date).format('DD/MM/YYYY')
+
+                        absences.push(`${formatDate}  ${(!(amLeave && pmLeave) ? (amLeave ? '- Matin' : '- Après-midi'): '')}`)
```

Comment:
( oui je sais c'est mon code ^^' )
```suggestion
                        absences.push(`${formatDate} ${(!(amLeave && pmLeave) ? (amLeave ? '- Matin' : '- Après-midi'): '')}`)
```